lib/logstash/outputs/elasticsearch.rb

```diff
@@ -425,43 +442,58 @@ def generate_jks cert_path
   def receive(event)
     return unless output?(event)
 
+    # block until we have not maxed out our 
+    # retry queue. This is applying back-pressure
+    # to slow down the receive-rate
+    @retry_queue_not_full.wait if @retry_queue.size > @retry_max_items
```

Comment:
condition variables wait should always check for met condition upon returning from wait

```ruby
@retry_queue_not_full.wait while @retry_queue.size > @retry_max_items
```